enhancements/dns/dns-operator-operand-logging-level.md

```diff
@@ -0,0 +1,230 @@
+}
+```
+
+This new field would allow a cluster administrator to specify the desired logging level specifically for the DNS Operator.
+
+Additionally, a new `LogLevel` of type [LogLevel](https://github.com/openshift/api/blob/master/operator/v1/types.go#L62) will be added for CoredDNS logging :
+
+```go
+// We will be enabling the CLASSES(https://github.com/coredns/coredns/tree/master/plugin/log#syntax) of coredns w.r.t to the LogLevel we have defined in openshift api.
+
+// So,
+
+// logLevel "Normal"  will enable errors class as we enable errors plugin by default log . { class error }
+
+// logLevel "Debug" will enable log . { class denial error }
+
+// logLevel "Trace" will enable  log . { class all }
+type DNSSpec struct {
+	
+	// logLevel describes the logging verbosity of the DNSController for CoreDNS.
+	//
+	// +optional
+	LogLevel LogLevel `json:"logLevel"`
+}
+
+```
+
+Both of these new APIs would be accompanied by appropriate `LogLevel` definitions:
+
+```go
+
+// LogLevel describes several available logging verbosity levels.
+// +kubebuilder:validation:Enum=Normal;Debug;Trace;TraceAll
+type LogLevel string
+
+var (
+	// Normal is the default.  Normal, working log information, everything is fine, but helpful notices for auditing or common operations.  In kube, this is probably glog=2.
+	Normal LogLevel = "Normal"
+
+	// Debug is used when something went wrong.  Even common operations may be logged, and less helpful but more quantity of notices.  In kube, this is probably glog=4.
+	Debug LogLevel = "Debug"
+
+	// Trace is used when something went really badly and even more verbose logs are needed.  Logging every function call as part of a common operation, to tracing execution of a query.  In kube, this is probably glog=6.
+	Trace LogLevel = "Trace"
+
+	// TraceAll is used when something is broken at the level of API content/decoding.  It will dump complete body content.  If you turn this on in a production cluster
+	// prepare from serious performance issues and massive amounts of logs.  In kube, this is probably glog=8.
+	TraceAll LogLevel = "TraceAll"
+)
+```
+
+### User Stories
+
+* As an OpenShift Cluster Administrator, I want to be able to raise the logging level of the DNS Operator and CoreDNS so that I can more quickly
+track down OpenShift DNS issues.
+
+* Some users recently added the new alert 'CoreDNS is returning SERVFAIL for X% of requests alert' to the recent updates of OCP.
+  Adding this prometheus alert is nice, but it would be more useful we can see which request are getting SERVFAIL response.
+  So we would to enable the log plugin for CoreDNS to log queries.
+  
+* Some user want to avoid use of tcpdump to see the queries and want log plugin to be enabled to log queries in coredns.
+
+
+### Implementation Details/Notes/Constraints [optional]
+
+
+### Risks and Mitigations
+
+Raising the logging verbosity for any component typically results in larger log files that grow quickly.
```

Comment:
If possible, each risk should have a mitigation.  For example, the godoc for the new API field could include a warning that setting `logLevel: Trace` will produce extremely verbose logs.

Reply:
@Miciah Do you mean add a comment in the API PR and recreate the go docs

Reply:
No, I mean the point of having a "Risks **and Mitigations**" section is that you'll describe the mitigation for each risk.